Spanner/src/Operation.php

```diff
@@ -339,13 +364,26 @@ public function executeUpdateBatch(
             ] + $this->mapper->formatParamsForExecuteSql($parameters, $types);
         }
 
+        if (!isset($options['transaction']['begin'])) {
+            $options['transactionId'] = $transaction->id();
+        }
+
         $res = $this->connection->executeBatchDml([
             'session' => $session->name(),
             'database' => $this->getDatabaseNameFromSession($session),
-            'transactionId' => $transaction->id(),
             'statements' => $stmts
         ] + $options);
 
+        if (
+            is_null($transaction->id()) &&
+            !empty($res['resultSets'][0]['metadata']['transaction']['id'])
+        ) {
+            // Get the transaction from array of ResultSets.
+            // ResultSet contains transaction in the metadata.
+            // @see https://cloud.google.com/spanner/docs/reference/rest/v1/ResultSet
+            $transaction->setId($res['resultSets'][0]['metadata']['transaction']['id']);
```

Comment:
Can it happen that `$res['resultSets']` produces an empty array?
The code execution wont enter this if condition, but can it lead to a failure?

Reply:
It always have this value. This is similar to Ruby as well. Refer: https://github.com/googleapis/ruby-spanner/blob/7ee6600d8dbad3596251747d1c6c052d5db3c736/google-cloud-spanner/lib/google/cloud/spanner/batch_update_results.rb

Reply:
`null` is the value already present as the `transactionId` at the time of initiation of `Transaction` object. So, there is not need to set that again.

Reply:
I meant it to be like this:
```suggestion
            $transaction->setId($res['resultSets'][0]['metadata']['transaction']['id'] ?? '');
```